tests/find_after_hooks.js

```diff
@@ -56,8 +56,12 @@ Tinytest.addAsync('issue #296 - after insert hook always finds all inserted', as
 Tinytest.addAsync('async find hook - after insert hook always finds all inserted', async function (test, next) {
   const collection = new Mongo.Collection(null)
 
-  collection.before.find(async (userId, selector) => {
-    await new Promise(resolve => setTimeout(resolve, 10)) // Simulate async operation
+  collection.before.find((userId, selector) => {
+    selector.removedAt = { $exists: false }
+    return true
+  })
+
+  collection.before.findOne((userId, selector) => {
```

Comment:
Actually after modifying the test, I think the hook should allow async operations :shrug: don't go about fixing your PR because I'm unsure if I'd wanna merge it.

Reply:
Backwards compatibility is a big thing, I'm working on a project where we need to have this async capability working. I've not had time to really solve this and mind you I've already poured in tons of time but hopefully this can be solved soon.

The whole package isn't straight forward, it's a mess of monkey patching and many workarounds :sweat_smile:

Reply:
After re-reading the source code and testing further, I doubt that we will be able to enable async `before.find` functions without major changes : `.find()` is a sync method in Meteor and returns a cursor synchronously. 
Trying to use an async `before.find` hook will imply the need to `await` every call of `.find()`, which isn't doable.

This is indeed breaking for some meteor 2 code

A possible alternative would be to implement this logic differently by remapping async cursor operations `['countAsync', 'fetchAsync', 'forEachAsync', 'mapAsync']` to enhanced async cursor operations. Those new hooks would be `before.count`, `before.fetch`, etc... 
I'm unsure this is doable and easy to use. 
 
I'm reopening PR #318 as the most suitable solution here